cli/command/image/build_test.go

```diff
@@ -18,6 +20,52 @@ import (
 	"golang.org/x/net/context"
 )
 
+func TestRunBuildResetsUidAndGidInContext(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Invalid test on Windows")
+	}
+	dest, err := ioutil.TempDir("", "test-build-context-dest")
+	require.NoError(t, err)
+	defer os.RemoveAll(dest)
+
+	fakeImageBuild := func(_ context.Context, context io.Reader, options types.ImageBuildOptions) (types.ImageBuildResponse, error) {
+		buffer := new(bytes.Buffer)
+		tee := io.TeeReader(context, buffer)
+		assert.NoError(t, archive.Untar(tee, dest, nil))
+
+		body := new(bytes.Buffer)
+		return types.ImageBuildResponse{Body: ioutil.NopCloser(body)}, nil
+	}
+
+	cli := test.NewFakeCli(&fakeClient{imageBuildFunc: fakeImageBuild})
+	dockerfile := bytes.NewBufferString(`
+		FROM alpine:3.6
+		COPY foo bar /
+	`)
+	cli.SetIn(command.NewInStream(ioutil.NopCloser(dockerfile)))
+
+	dir, err := ioutil.TempDir("", "test-build-context")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+
+	ioutil.WriteFile(filepath.Join(dir, "foo"), []byte("some content"), 0644)
+	err = os.Lchown(filepath.Join(dir, "foo"), 65534, 65534)
+	require.NoError(t, err)
```

Comment:
This test can use a Dockerfile from the context, instead of stdin. So `dockerfile` can be just the string, instead of a bytes.Buffer. and you don't need to set `cli.SetIn()`. Below you won't need the `options.dockerfileName = "-"` line either.

You can replace this block (line 45-53) with:

```go
dir := fs.NewDir(t, "test-build-context", 
    fs.WithFile("foo", "some content", AsUser(65534, 65534)),
    fs.WithFile("Dockerfile", dockerfile),
)
defer dir.Remove()
```

You'll have to update `vendor.conf` to use gotestyourself `v1.2.0` as `WithFile()` was fixed in that version to accept `PathOp`